sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -2048,11 +2048,57 @@ def df_set_column_overload(self, key, value):
     """
     Intel Scalable Dataframe Compiler User Guide
     ********************************************
+    Pandas API: pandas.DataFrame.setitem
+
+    Set data to a DataFrame by indexer.
 
     Limitations
     -----------
-    - Supported setting a columns in a non-empty DataFrame as a 1D array only.
+    - Supported setting a column in a DataFrame through private method ``df._set_column(key, value)``.
     - Unsupported change of the Parent DataFrame, returned new DataFrame.
```

Comment:
Maybe smth like this?
```suggestion
    - The Parent DataFrame that comes into jitted region as a function parameter cannot be changed. New DataFrame is returned in this case.
```